### PR TITLE
Task/161057 Create new Archive flow

### PR DESCRIPTION
**Pull request by pablocodeglia**
Closes task [AB#161057](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/161057)

Screenshots:

image
image
image
image
image

### CODE REVIEW

File: src/app/config/translations/en.ts [hunk 1]
```diff
@@ -654,6 +654,19 @@ export const locale = {
...
+            },
+            AI02_INNOVATION_ARCHIVED_TO_COLLABORATORS: {
+              title: `Innovation {{innovationName}} has been archived by its owner.`
+            },
+            AI03_INNOVATION_ARCHIVED_TO_ENGAGING_QA_A: {
+              title: `Innovation {{innovationName}} has been archived by its owner. `,
+              link: `You can view an archived copy of the innovation record.`
+            },
+            AI04_INNOVATION_ARCHIVED_TO_NA_DURING_NEEDS_ASSESSMENT: {
+              title: `Innovation {{innovationName}} has been archived by its owner during the needs assessment/reassessment process. You do not need to take any further action for this innovation.`
```
Comment (eduarda-meirinhos-bjss) [inline src/app/config/translations/en.ts:668]: I thought this part "assessment/reassessment" would be like {{assessmentType}} on notification NA01_INNOVATOR_SUBMITS_FOR_NEEDS_ASSESSMENT_TO_INNOVATOR and NA02_INNOVATOR_SUBMITS_FOR_NEEDS_ASSESSMENT_TO_ASSESSMENT

Comment (pablocodeglia) [reply src/app/config/translations/en.ts:668]: You are completely right!

File: src/modules/feature-modules/innovator/pages/innovation/manage/manage-overview.component.ts [hunk 2]
```diff
@@ -31,11 +32,19 @@ export class PageInnovationManageOverviewComponent extends CoreComponent impleme
...
+    ) {
+      this.isWaitingAssessment = true;
+    }
     if (this.innovation.status === InnovationStatusEnum.IN_PROGRESS) {
       this.isInProgressInnovation = true;
     } else if (this.innovation.status === InnovationStatusEnum.ARCHIVED) {
-      this.isInPause = true;
+      this.isArchived = true;
     }
+    console.log(this.isWaitingAssessment);
```
Comment (eduarda-meirinhos-bjss) [inline src/modules/feature-modules/innovator/pages/innovation/manage/manage-overview.component.ts:47]: Remove console.log

**Review by eduarda-meirinhos-bjss — APPROVED** [review]

File: src/modules/feature-modules/innovator/services/innovator.service.ts [hunk 3]
```diff
@@ -167,9 +167,9 @@ export class InnovatorService extends CoreService {
     );
   }
 
-  pauseInnovation(innovationId: string, message: string): Observable<{ id: string }> {
+  archiveInnovation(innovationId: string, message: string): Observable<{ id: string }> {
     const url = new UrlModel(this.API_INNOVATIONS_URL)
-      .addPath('v1/:innovationId/pause')
+      .addPath('v1/:innovationId/archive')
       .setPathParams({ innovationId });
     return this.http.patch<{ id: string }>(url.buildUrl(), { message }).pipe(
```
Comment (diogo-costa-bjss) [inline src/modules/feature-modules/innovator/services/innovator.service.ts:174]: Now this method doesn't return the id, now is a `204`. You can change the return type to `void`.

Comment (pablocodeglia) [reply src/modules/feature-modules/innovator/services/innovator.service.ts:174]: You indeed had already told me this. Thanks for the reminder!

File: src/modules/feature-modules/innovator/services/innovator.service.ts [hunk 4]
```diff
@@ -157,11 +157,11 @@ export class InnovatorService extends CoreService {
...
 
-  withdrawInnovation(innovationId: string, message: string): Observable<{ id: string }> {
+  withdrawInnovation(innovationId: string, message: string): Observable<void> {
     const url = new UrlModel(this.API_INNOVATIONS_URL)
       .addPath('v1/:innovationId/withdraw')
       .setPathParams({ innovationId });
-    return this.http.patch<{ id: string }>(url.buildUrl(), { message }).pipe(
+    return this.http.patch<void>(url.buildUrl(), { message }).pipe(
       take(1),
       map(response => response)
```
Comment (eduarda-meirinhos-bjss) [inline src/modules/feature-modules/innovator/services/innovator.service.ts:166]: No need for map when returning void

Comment (pablocodeglia) [reply src/modules/feature-modules/innovator/services/innovator.service.ts]: Also, had changed the wrong method. :flushed:
Already corrected.

**Pull request merged**